### ISSUE TITLE
[lustre/ui/menu] Support disabled state.

### ISSUE THREAD

**Issue by hayleigh-dot-dev** [lustre-labs/ui#26]
Individal menu items in a `<lustre-ui-menu />` should be able to be disabled. When an item is disabled it is skipped over in keyboard navigation<sup>1</sup> and clicking on it should not emit a `lustre-ui:select` event. Where the item type is a checkbox, clicking on the item should also not emit a `lustre-ui:toggle` event regardless of whether the checked value is controlled or uncontrolled.

We'll also want to make sure the element sets a `disabled` CSS custom state so users can style disabled items appropriately.

---

<sup>1</sup> Note that this runs _counter_ to the ARIA APG which says that disabled menu items must still be navigable by keyboard. Testing how both native menus and menus implemented in other libraries work mostly suggests that _skipping over disabled items_ is the most common interaction pattern and so we'll implement that as a principle of least surprise.
